REGRESSION(298383@main): [WPE] Bad touch pointer IDs

dos1 opened the pull request:
#### 7879cb55638ec765dc033dd9ba1a11bc3c789efa
<pre>
REGRESSION(298383@main): [WPE] Bad touch pointer IDs
<a href="https://bugs.webkit.org/show_bug.cgi?id=302845">https://bugs.webkit.org/show_bug.cgi?id=302845</a>

Reviewed by Carlos Garcia Campos.

298383@main has effectively reverted the change introduced in 287077@main in
what appears to be a mistake in conflict resolution. Afterwards, 300636@main
removed the now unused variable touchMinimumPointerId. Bring the change back
to re-fix the issue fixed by 287077@main.

* Source/WebCore/dom/wpe/PointerEventWPE.cpp:
(PointerEvent::PointerEvent): Brought touchMinimumPointerId back.

Canonical link: <a href="https://commits.webkit.org/303457@main">https://commits.webkit.org/303457@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/db70e9bb07263823b97815b9f80bad29a3f9604a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132332 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4827 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43370 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139847 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84278 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134202 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4953 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4586 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101160 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68437 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135278 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/3482 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118534 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81950 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/3365 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1158 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83072 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/112213 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36651 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142498 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4497 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37237 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109537 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4579 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3884 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109715 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27832 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3400 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114806 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/57773 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4551 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33173 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4383 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68001 "Built successfully") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4642 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4508 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

spenap commented on Source/WebCore/dom/wpe/PointerEventWPE.cpp:
```diff
@@ -71,9 +71,14 @@ Ref<PointerEvent> PointerEvent::create(const AtomString& type, const PlatformTou
+// According to the PointerEvents spec all active pointer ids have to be unique.
+// Libinput on Linux assigns the ids of the touchpoints starting at 0, but
+// the ids 0 and 1 are used for the pointer ids of mouse and pen/stylus.
+const unsigned touchMinimumPointerId = WebCore::mousePointerID + 1;
```
`WebCore::mousePointerID` is a static constexpr, so maybe this can be made one, too?

dos1 replied:
It is now, updated.

spenap replied:
Thanks! I am not a reviewer, but this looks good

dos1 committed:
REGRESSION(298383@main): [WPE] Bad touch pointer IDs
https://bugs.webkit.org/show_bug.cgi?id=302845

Reviewed by Carlos Garcia Campos.

298383@main has effectively reverted the change introduced in 287077@main in
what appears to be a mistake in conflict resolution. Afterwards, 300636@main
removed the now unused variable touchMinimumPointerId. Bring the change back
to re-fix the issue fixed by 287077@main.

* Source/WebCore/dom/wpe/PointerEventWPE.cpp:
(PointerEvent::PointerEvent): Brought touchMinimumPointerId back.

Canonical link: https://commits.webkit.org/303457@main
